libraries/core_libs/consensus/src/dag/dag_block_proposer.cpp

```diff
@@ -72,125 +64,137 @@ bool DagBlockProposer::proposeDagBlock() {
     return false;
   }
 
-  if (!isValidDposProposer(*proposal_period)) {
-    return false;
-  }
-
-  auto pk = key_manager_->getVrfKey(*proposal_period, node_addr_);
-  if (pk && *pk != vrf_pk_) {
-    LOG(log_er_) << "VRF public key mismatch " << *pk << " - " << vrf_pk_;
-    return false;
-  }
-
-  uint64_t max_vote_count = 0;
-  const auto vote_count = final_chain_->dposEligibleVoteCount(*proposal_period, node_addr_);
-  if (*proposal_period < kHardforks.magnolia_hf.block_num) {
-    max_vote_count = final_chain_->dposEligibleTotalVoteCount(*proposal_period);
-  } else {
-    max_vote_count = kValidatorMaxVote;
-  }
+  // Try to propose dag block for each wallet
+  // TODO: optimize
+  for (auto& node_dag_proposer_data : nodes_dag_proposers_data_) {
+    if (!isValidDposProposer(*proposal_period, node_dag_proposer_data.wallet.node_addr)) {
+      return false;
```

Comment:
I think this should be "continue" and not "return" so that if one of the validator is not valid proposer other still propose blocks. There is a couple of more places where "return" should be replaced with continue

Reply:
pls check this fix https://github.com/Taraxa-project/taraxa-node/pull/3021/commits/3dcb0e8ceca79209bbe79d6f19dad2ae06824655